Model/Sync/CollectionsProducts/Processor.php

```diff
@@ -2,10 +2,352 @@
+                $collectionsProductsEntitiesToSync = $this->enrichCollectionsProductsDataWithYotpoIds($collectionsProductsEntitiesToSync, $collectionsProductsCategoriesIds, $collectionsProductsProductIds);
+
+                foreach ($collectionsProductsEntitiesToSync as $collectionProductEntityToSync) {
+                    $collectionYotpoId = $collectionProductEntityToSync['category_yotpo_id'];
+                    $productYotpoId = $collectionProductEntityToSync['product_yotpo_id'];
+
+                    if (!$collectionYotpoId) {
+                        $magentoCategoryId = $collectionProductEntityToSync['magento_category_id'];
+                        $collectionYotpoId = $this->syncCollectionAndGetCreatedYotpoId($magentoCategoryId, $storeId);
+                        if ($collectionYotpoId == null) {
+                            $this->collectionsProductsService->updateCollectionsProductsSyncData($storeId, $collectionProductEntityToSync);
+                            continue;
+                        }
+                    }
+
+                    if (!$productYotpoId) {
+                        $magentoProductId = $collectionProductEntityToSync['magento_product_id'];
+                        $productYotpoId = $this->syncProductAndGetCreatedYotpoId($magentoProductId, $storeId);
+                        if ($productYotpoId == null) {
+                            $this->collectionsProductsService->updateCollectionsProductsSyncData($storeId, $collectionProductEntityToSync);
+                            continue;
+                        }
+                    }
+
+                    $requestEndpoint = $this->coreConfig->getEndpoint($requestEndpointKey, [$collectionIdRequestParamString], [$collectionYotpoId]);
+                    $collectionProductDataToSync = $this->prepareCollectionProductToSync($collectionProductEntityToSync);
+                    $collectionProductDataToSync['entityLog'] = $this::LOGGER_LOG_ENTITY_FILE;
+                    $isCollectionProductDeletedInMagento = (bool) $collectionProductEntityToSync['is_deleted_in_magento'];
+
+                    $response = $this->syncCollectionProductToYotpo($requestEndpoint, $collectionProductDataToSync, $isCollectionProductDeletedInMagento);
+
+                    if ($response->getData('status') == $this->coreConfig::CONFLICT_RESPONSE_CODE) {
+                        $this->setAsSuccessfulCollectionProductSync($storeId, $collectionProductEntityToSync);
+                        continue;
+                    }
+
+                    $isResyncableStatusCode = $this->coreConfig->canResync($response->getData('status'));
+                    if ($response->getData('is_success')) {
+                        $this->setAsSuccessfulCollectionProductSync($storeId, $collectionProductEntityToSync);
+                    } elseif (!$isResyncableStatusCode) {
+                        $this->setAsSuccessfulCollectionProductSync($storeId, $collectionProductEntityToSync);
+                    }
```

Comment:
We're keeping here a loophole for 404 exceptions.
Let's discuss this separately.

Reply:
Changed it to be set as synced as we discussed offline, made it nicer by moving it and the can resync method to a private function:
 https://github.com/YotpoLtd/magento2-module-core/pull/163/files#diff-c0e0b55174eac17a090995cd7615a2cff671e8b81d9711995cfdb5538ec3d3dcR348